### ISSUE TITLE
Feature request: entity whitelist (inverse of ignore rule)

### ISSUE THREAD

**Issue by danielrosehill** [dummylabs/thewatchman#121]
I'm configuring Watchman to notify of unavailable lights (specifically).

Currently the only way for me to achieve this is to:

1: Run an offline entities report
2: Add all the notification groups I don't want to hear about by adding them as ignore options (scene.*, sensor.* etc).

It would be great if there were the inverse option of ignore -- a whitelist (only report on entities matching these conditions).

Then I could just add: light.* to achieve the same functionality.

**Comment by nbetcher** [comment]
In theory there should be a way since (for some bizarre reason, versus `re.match()`) the author is using `fnmatch()` on each _entity_ and _service_ while building the list and, because of that, Unix filenaming globbing like this SHOULD work (but for some reason, doesn't):

```
!(notify.pushover)
+(!(notify.pushover))
?(!(notify.pushover))
@(!(notify.pushover))
```
_(That assumes the only whitelisted entity or service would be [notify.pushover](url) and everything else gets blacklisted.)_

Ideally it would be switched to `re.match()` since these in particular are not filenames in the first place (`fnmatch()` = _**File Name** Match_), then we could take the guess-work out of it and "simply" put:

^((?!notify\.pushover).)*$

Or yes, add a whitelist-only mode. ;)

**Comment by dummylabs** [comment]
Duplicate of #20.

**Comment by nbetcher** [comment]
> Duplicate of #20.

I wouldn't exactly call it a duplicate, at least my comment. You're using `fnmatch()` which is described (and named) as being for filename matching. While Regular Expressions are more sophisticated and don't respond to `*` the same way as `fnmatch()` would,
I don't believe there's any precedent in Home Assistant to accept Unix globbing patterns over Regular Expressions.

At the very least it would be nice to have the option to use Regular Expressions since they provide more flexibility to your advanced users, but more importantly could get you off the hook for a plethora of enhancements you'd otherwise have to manually code yourself. (e.g. white-list only mode)

If you wanted to make it transparent to users, you could certainly require that regular expressions be prefixed and suffixed with `/` like JavaScript requires, like so:

`/^sensor\.whos_your_daddy$/`

More of an enhancement request, ultimately, but it certainly solves a lot of your bugs, both current and future state. If you'd like, I can open another issue as an enhancement.

**Comment by dummylabs** [comment]
@nbetcher Yes, I understand your point. I have marked the initial request as a duplicate because, in my opinion, both tickets share the same intention: to have the ability to restrict the list of entities Watchman should track. Your idea is excellent, and I’ll keep it in mind for future consideration if we decide to implement a whitelist.

**Comment by dummylabs** [comment]
I’ve re-examined your request, and it seems to me that using Watchman to monitor disappearing entities isn’t quite its intended purpose—it’s not directly related to configuration issues. It would be much simpler to set up a group and monitor their state via automation. I hope you understand 🤝.
